src/ccache.cpp

```diff
@@ -3434,12 +3426,10 @@ cc_process_args(struct args* args,
   // Add flags for dependency generation only to the preprocessor command line.
   if (generating_dependencies) {
     if (!dependency_filename_specified) {
-      char* base_name = remove_extension(output_obj);
-      char* default_depfile_name = format("%s.d", base_name);
-      free(base_name);
+      std::string default_depfile_name = Util::change_extension(output_obj, ".d");
```

Comment:
I you have the possibility, consider installing clang-format and running `make format`.